### ISSUE TITLE
Using FEC to compile a property setter with an enum value sets the property to random values

### ISSUE THREAD

**Issue by CodingGorilla** [dadhi/FastExpressionCompiler#172]
I have the following code that I use to do some simple mapping between instances:
```
public static Action<TEntity, TProperty> CreateSetter<TProperty>(Expression<Func<TEntity, TProperty>> property)
{
    var propertyInfo = GetProperty(property);

    var instance = Expression.Parameter(typeof(TEntity), "instance");
    var parameter = Expression.Parameter(typeof(TProperty), "param");

    var body = Expression.Call(instance, propertyInfo.GetSetMethod(), parameter);
    var parameters = new[] { instance, parameter };

    return Expression.Lambda<Action<TEntity, TProperty>>(body, parameters).CompileFast();
}
```

This works for most properties, but when I use it with one (that I know of) property which has an enum type, the value ends up being set to random integers that are not a part of the actual enum.  Removing FEC and just using `.Compile()` seems to solve the problem.

**Comment by dadhi** [comment]
Hi there, the version of FEC?

**Comment by CodingGorilla** [comment]
Sorry, version 1.10.1, in an .NET Core 2.1 dll that is hosted in an ASP.NET MVC Core 2.1 web api.

**Comment by dadhi** [comment]
Try to copy `FastExpressionCompiler.cs` from the repo directly to your project. Check if problem is still there. 

The new version (2.0) will be out soon with many fixes related to enums as well. So you may check in advance.

**Comment by CodingGorilla** [comment]
That did seem to solve the problem.  Is that the 2.0 version of the code?  And if so, do you know when (roughly) you plan to release it?

**Comment by dadhi** [comment]
Yes, it is 2.0. Preview will be creleased Today. Stable one by the end-of-the-week. But you may just use the file until then.

**Comment by dadhi** [comment]
v2.0.0-preview-01 is [out](https://www.nuget.org/packages/FastExpressionCompiler/2.0.0-preview-01). Please check it out.
